File: backend/src/main/java/ipeps/pwd/wallet/module/company/service/CompanyService.java
package ipeps.pwd.wallet.module.company.service;


import ipeps.pwd.wallet.module.company.entity.Company;
import ipeps.pwd.wallet.module.company.entity.payload.CompanyCreatePayload;
import ipeps.pwd.wallet.module.company.entity.payload.CompanyUpdatePayload;

import java.util.List;
import java.util.UUID;

public interface CompanyService {
    public List<Company> list();

    public Company detail(UUID companyId);

    public Company create(CompanyCreatePayload payload);

    public Company  update(CompanyUpdatePayload payload);

    public boolean delete(UUID companyId);
}

File: WalletApp/src/app/shared/service/crud/company.service.ts
import {Observable} from "rxjs";
import {Injectable} from "@angular/core";
import {ApiService} from "@shared/service";
import {ApiResponse, CrudServiceInterface} from "@shared/model";
import {map} from "rxjs/operators";
import {Company} from "@shared/model/dto/company.interface";
import {CompanyCreatePayload} from "@shared/model/payload/create/CompanyCreatePayload.interface";
import {
  CompanyUpdatePayload
} from "@shared/model/payload/update/CompanyUpdatePayload.interface";

@Injectable({
  providedIn: 'root'
})
export class CompanyService extends ApiService implements CrudServiceInterface {

  list = () : Observable<Company[]> =>  {
    return this.get(`company/list`).pipe(
      map((response: ApiResponse) => {
        return response.result ? response.data as Company[] : [];
      })
    )
  }

  detail = (id: string | number) : Observable<Company> => {
    return this.get(`company/detail/${id}`).pipe(
      map((response: ApiResponse) => {
        return response.result ? response.data as Company : {} as Company;
      })
    )
  }

  create(payload: CompanyCreatePayload): Observable<boolean> {
    return this.post(`company/create/`, payload).pipe(
      map((response: ApiResponse) => {
        return response.result;
      })
    )
  }

  update(payload: CompanyUpdatePayload): Observable<boolean> {
    return this.put('company/update/', payload).pipe(
      map((response: ApiResponse) => {
        return response.result
      })
    )
  }

  remove(id: string | number): Observable<boolean> {
    return this.delete(`company/delete/${id}`).pipe(
      map((response: ApiResponse) => {
        return (response.result)
      })
    )
  }
}
